mypy.ini

```diff
@@ -0,0 +1,9 @@
+# mypy.ini
+[mypy]
+check_untyped_defs = True
+disallow_untyped_defs = False
+ignore_missing_imports = True
+disable_error_code = union-attr, attr-defined, typeddict-unknown-key, no-any-return, return, assignment
```

Comment:
We should avoid bulk disabling error codes. Mainly this will avoid scenarios where we introduce a new typing issue that is not caught because the error is globally disabled. Instead of using `disable_error_code`, I'm thinking we:

1. Try to fix the actual typing issue
2. If it is not possible or very impractical to fix the typing issue, we can add an inline ignore error code.

At some point, we should even consider adding the [`strict`]https://mypy.readthedocs.io/en/stable/existing_code.html#introduce-stricter-options) option, but I think that can be a follow up as that would likely require more changes and could be even more finicky to get working.